### DISCUSSION TITLE
Add searchable to Select Filter

### DISCUSSION THREAD

**Discussion by waadmawlood** [laravel/nova-issues#6309] [Ideas & Feature Requests]
Laravel Version: 10
Nova Version: 4.33.0
PHP Version: 8.2.17
Operating System and Version: Ubuntu 22.04
Browser type and version: Latest on all browsers tested

### Description:
I need to add searchable for select filter (select-filter) because the list data is big
![IMG_20240326_151517](https://github.com/laravel/nova-issues/assets/44348636/78bed9fe-26d9-4692-bb30-fd48e91ffa9f)

**Comment by webard** [comment]
Just use `searchable()` on `Select` field, like documentation said: https://nova.laravel.com/docs/resources/fields.html#select-field

**Comment by waadmawlood** [comment]
> Just use `searchable()` on `Select` field, like documentation said: https://nova.laravel.com/docs/resources/fields.html#select-field

I am not using `Select Field` but I am using `Select Filter` [https://nova.laravel.com/docs/filters/defining-filters.html#select-filters](https://nova.laravel.com/docs/filters/defining-filters.html#select-filters)

**Comment by crynobone** [comment]
Searchable Select Filter may not be able to improve performance (or make it slower) as all the options will be loaded on the initial request similar to Select. The only improvement it would offer is the ability to search available options.

**Reply by waadmawlood** [reply]
Well, if it affects the efficiency and speed because it is linked to the database, then we do not want that. Rather, make the search through JavaScript. The browser will do the job sufficiently, meaning the data is already available, but the search is through JavaScript.

**Reply by nastoychev** [reply]
@crynobone We appreciate your efforts! Since you already have the Vue component from the searchable field, it seems like it wouldn't take too much effort to implement this. Thank you!

**Comment by waadmawlood** [comment]
Well, if it affects the efficiency and speed because it is linked to the database, then we do not want that. Rather, make the search through JavaScript. The browser will do the job sufficiently, meaning the data is already available, but the search is through JavaScript. example (Select2 js)

**Comment by Gokalex** [comment]
I also need this, i have some filters with 100s of values and i need to quickly be able to search by typing parts of it, is there any update on when or if this feature will be implemented?
I found some mods that don't work on nova4 (guess they are for nova3)
https://github.com/Fontech/nova-searchable-select-filter
https://github.com/angauber/nova-select2-filter

Can they be modified to work with nova4?

Thanks

**Comment by ekateiva** [comment]
We also need this functionality. Especially, that `Select::make()->filterable()->searchable()` already has tis functionality. 
It would be really cool to have the same functionality for `select-filter` component.

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.

**Reply by arall** [reply]
Is there any way to make a custom written filter searchable? Or this is only achievable with `Select::make()->filterable()->searchable()`?

**Reply by nastoychev** [reply]
You CAN create custom filter with filterable option. 

The behaviour and the UI will be a bit different.

**Reply by arall** [reply]
I would love to know how, somehow can't find it in the docs.
